src/main/java/bob/Bob.java

```diff
@@ -0,0 +1,209 @@
+package bob;
+
+import bob.exception.DirectoryNotFoundException;
+import bob.exception.FileNotFoundException;
+
+import java.io.File;
+import java.util.Objects;
+
+import javafx.animation.PauseTransition;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+/**
+ * A chatbot that stores, displays and alters the user's task list based on the user input.
+ */
+public class Bob extends Application{
+    /** Storage object that deals with loading tasks from the file and saving tasks in the file */
+    private Storage storage;
+
+    /** List of user tasks */
+    private TaskList tasks;
+
+    /** Ui object that deals with interactions with the user */
+    private Ui ui;
+
+    /** Parser object that deals with making sense of the user commands */
+    private Parser parser;
+
+    /** Whether the data directory is already present in the user computer */
+    boolean isDirectoryPresent = true;
+
+    /** Whether the bob.txt file is already present within the data directory in the user computer */
+    boolean isBobFilePresent = true;
+
+    /** UI element that allows users to scroll up and down the stage to view more content */
+    private ScrollPane scrollPane;
+
+    /** UI element that lays out its children in a single vertical column */
+    private VBox dialogContainer;
+
+    /** UI element that allows users to type in their input */
+    private TextField userInput;
+
+    /** UI element that allows Bob to take in the user input when the user clicks on it */
+    private Button sendButton;
+
+    /** UI element that contains all the nodes to be shown in the GUI */
+    private Scene scene;
+
+    /** Image representing the user in the chat */
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+
+    /** Image representing Bob in the chat */
+    private Image bob = new Image(this.getClass().getResourceAsStream("/images/DaBob.jpeg"));
+
```

Comment:
Wow, great that you described all your variables, definitely improves readability!